File: paper_artifacts/experiments/llr9/artifacts/sources/kimi27sglang-c-skills/fuse_move_ifs__submitted.c
#include <stdint.h>
#include <stdlib.h>
#include <math.h>
#include <omp.h>

void fuse_move_ifs_fp64(double* a,
                        double* b,
                        double* cond,
                        double* src,
                        int64_t K,
                        int64_t LEN_2D,
                        uint8_t* workspace,
                        int64_t workspace_bytes)
{
    (void)workspace;
    (void)workspace_bytes;

    const int64_t n = LEN_2D;

    if (K > 0) {
        #pragma omp parallel for schedule(static)
        for (int64_t i = 0; i < n; ++i) {
            const int64_t row = i * n;
            if (cond[i] > 0.0) {
                #pragma omp simd
                for (int64_t j = 0; j < n; ++j) {
                    a[row + j] = src[row + j] * 2.0;
                }
            }
            #pragma omp simd
            for (int64_t j = 0; j < n; ++j) {
                b[row + j] = src[row + j] + 1.0;
            }
        }
    } else {
        #pragma omp parallel for schedule(static)
        for (int64_t i = 0; i < n; ++i) {
            if (cond[i] > 0.0) {
                const int64_t row = i * n;
                #pragma omp simd
                for (int64_t j = 0; j < n; ++j) {
                    a[row + j] = src[row + j] * 2.0;
                }
            }
        }
    }
}

File: paper_artifacts/experiments/llr8/artifacts/sources/kimi27sglang-fortran/fuse_move_ifs__submitted.f90
module fuse_move_ifs_m
  use iso_c_binding, only: c_double, c_int64_t
  implicit none
contains
  subroutine fuse_move_ifs_fp64(a, b, cond, src, K, LEN_2D) bind(c, name="fuse_move_ifs_fp64")
    implicit none
    integer(c_int64_t), value, intent(in) :: K, LEN_2D
    real(c_double), intent(inout) :: a(LEN_2D,LEN_2D), b(LEN_2D,LEN_2D)
    real(c_double), intent(in) :: cond(LEN_2D), src(LEN_2D,LEN_2D)
    integer(c_int64_t) :: i, j, n
    n = LEN_2D
    !$omp parallel private(i,j)
    if (K > 0_c_int64_t) then
      !$omp do schedule(nonmonotonic:guided) nowait
      do i = 1_c_int64_t, n
        if (cond(i) > 0.0_c_double) then
          !$omp simd
          do j = 1_c_int64_t, n
            a(j,i) = src(j,i) * 2.0_c_double
          end do
          !$omp simd
          do j = 1_c_int64_t, n
            b(j,i) = src(j,i) + 1.0_c_double
          end do
        else
          !$omp simd
          do j = 1_c_int64_t, n
            b(j,i) = src(j,i) + 1.0_c_double
          end do
        end if
      end do
      !$omp end do
    else
      !$omp do schedule(nonmonotonic:guided) nowait
      do i = 1_c_int64_t, n
        if (cond(i) > 0.0_c_double) then
          !$omp simd
          do j = 1_c_int64_t, n
            a(j,i) = src(j,i) * 2.0_c_double
          end do
        end if
      end do
      !$omp end do
    end if
    !$omp end parallel
  end subroutine fuse_move_ifs_fp64
end module fuse_move_ifs_m
